CONTRIBUTING.md

```diff
@@ -1,65 +1,146 @@
+## Requesting Features
 
-    - **`utilites`**: A variety of helpers, decorators and utilities for working with Flux.
+For any feature requests, please use the [issue tracker](https://github.com/sage/carbon/issues). When you raise the feature request, please provide as much information as possible to describe the feature required, as well as examples of how you would like the feature to work.
+
+## Defining Features
+
+For features that you have a proposed solution for, please write an [RFC](./rfcs/README.md).
 
-    - **`style-config`**: Global styles and config for the Carbon library.
+## Automated issue management
+
+We use a bot to help us manage issues. The bot automatically labels issues where there has been no activity for 365 days as `stale`. We then consider the issue for closure. 
 
 ## Testing
 
 * Carbon has a 100% coverage policy. Testing is done using [Jest](https://facebook.github.io/jest/). We use the [Enzyme](https://github.com/airbnb/enzyme) testing utility for interacting with components in tests.
 * Legacy code used React Testutils - we are currently in the process of migrating to 100% Enzyme usage.
 * New tests need to be written using Enzyme - it's ok for a Testutils and Enzyme to be mixed in a legacy code file.
 
-## Submitting Pull Requests
-
-When submitting a pull request, please ensure your branch meets the following criteria:
-
-* It is a passing build (`gulp test --build` returns success).
-* The code is fully tested and has full coverage.
-* The code follows our coding guidelines (TBC).
-* The code does not introduce unnecessary dependencies (no jQuery!).
-* Ensure descriptive `.yml` file is added to the `./change/next` directory.
+## Pull Request Guidelines
+Before submitting a pull request, check the [issue tracker](https://github.com/sage/carbon/issues) to see if the feature or bug has already been discussed. If it has, check with us before beginning work on it to avoid duplicated effort. If no issue has been raised, please raise one first before beginning work. You can then link back to this when submitting a pull request to address it.
+
+Pull requests can be used in conjunction with two workflows:
+### The **Feature Branch** workflow 
+If you are a Carbon maintainer you should develop your feature in a branch of the Carbon repository. To merge into `master` you must first open a pull request with your feature branch as the source and `master` as the destination.
+
+### The **Forking** workflow 
+If you are a third party contributor you should push your feature to your own public repository, this will be the pull request source and the destination will typically be the Carbon repo `master` branch.
+
+Whichever workflow you use when you create the pull request you should include a detailed description of the feature you are adding and ensure you have included information on how to setup and QA your new feature or bug-fix.
+
+Your branch should meets the following criteria:
+
+1. Release notes are included.
```

Comment:
Strictly speaking we dont write release notes, but it's worth mentioning git message format here.

Also, theres often confusion around working on a feature branch and the message. e.g. if you break something on a feature branch there is no need to do a `fix` commit. Instead they should squash the fix with the commit that broke it.

If they don't do this they will end up with a patch release for a bug that was never released.